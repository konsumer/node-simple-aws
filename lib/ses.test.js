/* global describe, it, expect, jest */

const { SES } = require('..')

const ses = new SES()

describe('ses', () => {
  it('should have unit-tests', () => {
    expect(1 + 1).toBe(2)
  })

  it('should be able to send an email', async () => {
    ses.client.sendEmail = jest.fn((params, cb) => {
      expect(params).toMatchSnapshot()
      cb(null, true)
    })
    await new Promise((resolve) => {
      ses.sendEmail(
        'konsumer@jetboystudio.com',
        'konsumer@jetboystudio.com',
        'TESTING',
        'This is a test.',
        '<p>This is a test.</p>',
        (err, res) => {
          expect(ses.client.sendEmail.mock.calls.length).toBe(1)
          expect(err).toBe(null)
          expect(res).toBe(true)
          resolve()
        }
      )
    })
  })
})
