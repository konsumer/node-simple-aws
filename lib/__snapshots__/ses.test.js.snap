// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ses should be able to send an email 1`] = `
Object {
  "Destination": Object {
    "ToAddresses": Array [
      "konsumer@jetboystudio.com",
    ],
  },
  "Message": Object {
    "Body": Object {
      "Html": Object {
        "Data": "<p>This is a test.</p>",
      },
      "Text": Object {
        "Data": "This is a test.",
      },
    },
    "Subject": Object {
      "Data": "TESTING",
    },
  },
  "Source": "konsumer@jetboystudio.com",
}
`;

exports[`ses should be able to send an email, with a promise 1`] = `
Object {
  "Destination": Object {
    "ToAddresses": Array [
      "konsumer@jetboystudio.com",
    ],
  },
  "Message": Object {
    "Body": Object {
      "Html": Object {
        "Data": "<p>This is a test.</p>",
      },
      "Text": Object {
        "Data": "This is a test.",
      },
    },
    "Subject": Object {
      "Data": "TESTING",
    },
  },
  "Source": "konsumer@jetboystudio.com",
}
`;
